src/ImageSharp/Formats/Jpeg/Components/Decoder/ArithmeticStatistics.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+
+namespace SixLabors.ImageSharp.Formats.Jpeg.Components.Decoder
+{
+    internal class ArithmeticStatistics
```

Comment:
Can this be written as
```c#
internal unsafe struct ArithmeticStatistics
{
    private fixed byte statistics[256];

    public ArithmeticStatistics(bool dc, int identifier)
    {
        this.IsDcStatistics = dc;
        this.Identifier = identifier;
    }

    public bool IsDcStatistics { get; private set; }

    public int Identifier { get; private set; }

    public ref byte GetReference() => ref this.statistics[0];

    public ref byte GetReference(int offset) => ref this.statistics[(uint)offset];

    public void Reset()
    {
        Unsafe.InitBlockUnaligned(ref this.GetReference(), 0x00, (uint)(this.IsDcStatistics ? 64 : 256));
    }
}
```
?
* avoids the additional allocation for each stats-object
* downside: size is _fixed_ to 256, so potentially the list will be larger
* [codegen](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNXjxldskht1zYAZjAa4MUDmAwMAglF4YAFgL5hZGbH0vLcNADeNAyhDGJOCj7mJrwIMAAmDMAAnhjmlj68fmC4ANqkAKxIALoa1CFhxADMDk6u7speWTm4ABTAEBAGCWBoDNp2vAkwPLyxMgCUlaHB1GELDK7ZLACSuAAint6+HrgMALwMveWLYcu4ayNjE1CHA9d8t6dhAL6a81W1nd0M61vNXb+BiBBgAcxgGDU4Ui0QskOh7wqn1CNQGPD+j3GvBkIPBCJhvCi6XhUIYSJmTFqsBMyTS5gA4pC2DAzLAuCo2pNDgA+Bg0pYuFaZIG5PJ0MofBZogWpElMjAstmjTmDBgQEwmXCQ7kHPkCi4sEXZPZ5NocQaTDVayGS5HS2rEFAMFnajBcylzM6hACqXGMZjWXGcACFJBAwABrP36XhgriJNoGoWXBVKmQqmBc/p0BB0Oj9c2WtqG/7bFp7BgAfgYqAYIAYRSQk0mL1CSNeQA) is better than [with the class](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKAEsA7DGKe7AGwAIyOBBKWjAAt8MDLTABlDNjG4xYXDQ7KOAbyUrNAB34A3GTA6xsAEwj02ATw7BLTANoBdDnJm0543AG4aGzcuIAZl5+IRF5KTcPBQAKYAgIThMwNA4GDDSTGEZaADNaZgBKP391an8KjiF3ADoASVwAEUlpWXlcDgBeDmSfcsrNatx6rJz85i7M7LFxqD6BwcFa1zbPSeSOAH4OehgAdxs7GHtUZxAd/cOHUgBWJEd5yoBfX36KoJsEzgbmyNWFNQcADmoi8HB0tH0TBcoI4L2oJU0H3SHDqoxmBSggJBGDBEKhhlwsPhiJUH1guSuhgA4qIAEowXLMbJgGAxQpdAB8RkZVSWwxW7na9gADA9XgNybzbNDaRgGUzYPRWTEURBcrkiRgOZ1uRS+ctWkLPPZ1ZrROKEW9/B9iCgOAyteyuQaBUbosMeLgJFpsPR2TUAMJsGDYKDsx7KJ5AA=) especially for `GetReference` and `Reset` (which are the hottest methods here?)